app/src/main/java/com/nickytoolchick/agraph/MainActivity.kt

```diff
@@ -16,22 +16,33 @@ import kotlinx.serialization.json.Json
 class MainActivity : AppCompatActivity() {
 
     private var chartOptions = ChartOptions()
+    private var datasetOptions = DatasetOptions()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
         binding.configureChartButton.setOnClickListener {
-            val configIntent = Intent(this, ChartOptionsActivity::class.java)
-            configIntent.putExtra("stableChartOptions", Json.encodeToString(chartOptions))
-            resultLauncher.launch(configIntent)
+            val charOptionsIntent = Intent(this, ChartOptionsActivity::class.java)
+            charOptionsIntent.putExtra("stableChartOptions", Json.encodeToString(chartOptions))
+            chartOptionsResultLauncher.launch(charOptionsIntent)
+        }
+        binding.configureDatasetButton.setOnClickListener {
+            val datasetOptionsIntent = Intent(this, DatasetOptionsActivity::class.java)
+            datasetOptionsIntent.putExtra("stableDatasetOptions", Json.encodeToString(datasetOptions))
+            datasetOptionsResultLauncher.launch(datasetOptionsIntent)
+        }
+    }
+
+    private val chartOptionsResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            chartOptions = Json.decodeFromString(result.data?.getStringExtra("newChartOptions")!!)
         }
     }
 
-    private val resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+    private val datasetOptionsResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
         if (result.resultCode == Activity.RESULT_OK) {
-            chartOptions = Json.decodeFromString(result.data?.getStringExtra("result")!!)
-            Log.d("chart", chartOptions.horizontalStep.toString())
+            datasetOptions = Json.decodeFromString(result.data?.getStringExtra("newDatasetOptions")!!)
```

Comment:
Do not use hardcoded identifiers.

Transferring an extra data identifier to a separate Enum class or object can be a good practice. This can ensure the correctness of the identifier and help avoid typos or errors in the use of identifiers for the correct operation of the program

* Example:
```kotlin
object Exstras{
val NEW_DATASET_OPTIONS = "newDatasetOptions"
}
```